### PR TITLE
Move compiling assets to build time

### PR DESCRIPTION
**Pull request by bvallelunga**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,5 +48,5 @@
     "test": "./run-tests-travis.sh",
     "install": "npm prune; npm shrinkwrap"
   },
-  "version": "0.5.10"
+  "version": "1.0.0"
```
Comment (Piccirello) [inline package.json:51]: If we're bumping to `1.0` can we also switch to `package-lock`?

File: lib/piler.coffee [hunk 2]
```diff
@@ -301,11 +305,6 @@ class PileManager
     @app = app
     @server = server
 
-    listener = if server? then server else app
-    listener.on "listening", =>
-      @pileUp()
```
Comment (Piccirello) [inline lib/piler.coffee:306]: How does `pileUp` get called now?

Comment (bvallelunga) [reply lib/piler.coffee:306]: We call it manually in the server repo. I updated the readme to illustrate.

**Comment by bvallelunga** [comment]
> I don't full understand the difference. It looks like the old code synchronously compiled all files.

It it is async code but not using promises. All I am doing is making sure that for loop respects that is async and wrapping the function in a promise.

File: npm-shrinkwrap.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@dopplerhq/piler",
-  "version": "0.5.10",
+  "version": "1.0.0",
```
Comment (bvallelunga) [inline npm-shrinkwrap.json:3]: @Piccirello package-lock is updated.

File: package.json (hunk 1 above)
Comment (bvallelunga) [reply package.json:51]: yup ill updated.

File: npm-shrinkwrap.json (hunk 3 above)
Comment (Piccirello) [reply npm-shrinkwrap.json:3]: I don't think it got added to the repo

Comment (bvallelunga) [reply npm-shrinkwrap.json:3]: Just added it, it was ignored because of the `.gitignore`.

**Pull request merged**